src/python-frontend/python_converter.cpp

```diff
@@ -641,6 +766,29 @@ bool python_converter::is_constructor_call(const nlohmann::json &json)
   return is_ctor_call;
 }
 
+void python_converter::update_instance_from_self(
+  const std::string &class_name,
+  const std::string &func_name,
+  const std::string &obj_symbol_id)
+{
+  std::string self_id =
+    create_symbol_id() + "@C@" + class_name + "@F@" + func_name + "@self";
+
+  auto self_instance = instance_attr_map.find(self_id);
+  if (self_instance != instance_attr_map.end())
+  {
+    std::vector<std::string> &attr_list = instance_attr_map[obj_symbol_id];
```

Comment:
Is the order of elements in `attr_list` important? If not, you could consider using `std::unordered_map` and `std::unordered_set` for faster lookups.